### PR TITLE
fix: issue #107

### PR DESCRIPTION
**Pull request by christiangda**
This should fix issue #107 , but it is hard to test mocking the Google API method https://github.com/slashdevops/idp-scim-sync/issues/107

### CODE REVIEW

File: pkg/google/google.go [hunk 1]
```diff
@@ -76,59 +76,74 @@ func NewDirectoryService(svc *admin.Service) (*DirectoryService, error) {
...
+		err := ds.svc.Users.List().Customer("my_customer").Fields(listUsersRequiredFields).Pages(ctx, func(users *admin.Users) error {
 			u = append(u, users.Users...)
 			return nil
 		})
+		if err != nil {
+			return nil, err
+		}
 	}
-	return u, err
+	return u, nil
```
Comment (laurentdelosieresfact) [inline pkg/google/google.go:108]: @christiangda why not leaving it as such ? "return u, err" ?

File: pkg/google/google.go [hunk 2]
```diff
@@ -76,59 +76,74 @@ func NewDirectoryService(svc *admin.Service) (*DirectoryService, error) {
...
+		err := ds.svc.Groups.List().Customer("my_customer").Fields(groupsRequiredFields).Pages(ctx, func(groups *admin.Groups) error {
 			g = append(g, groups.Groups...)
 			return nil
 		})
+		if err != nil {
+			return nil, err
+		}
 	}
-	return g, err
+	return g, nil
```
Comment (laurentdelosieresfact) [inline pkg/google/google.go:146]: @christiangda why not leaving it as such ? "return u, err" ?

File: pkg/google/google.go [hunk 3]
```diff
@@ -173,8 +188,11 @@ func (ds *DirectoryService) ListGroupMembers(ctx context.Context, groupID string
 		}
 		return nil
 	})
+	if err != nil {
+		return nil, err
+	}
 
-	return m, err
+	return m, nil
```
Comment (laurentdelosieresfact) [inline pkg/google/google.go:195]: @christiangda why not leaving it as such ? "return u, err" ?

**Review by laurentdelosieresfact — CHANGES_REQUESTED** [review]

**Comment by laurentdelosieresfact** [comment]
Hello @christiangda ,

We have been running the idp-scim with the changes over the past 2 months, and we have not had any issues, everything is running smoothy. Could you merge the PR ? :pray.

Thank you in advance,

**Comment by snavarro-factorial** [comment]
+1 From my side too; we had no issues of groups being deleted and recreated randomly.

**Pull request merged**